lib/macho/load_commands.rb

```diff
@@ -400,6 +400,32 @@ def initialize(view, cmd, cmdsize, segname, vmaddr, vmsize, fileoff,
       @flags = flags
     end
 
+
+    # All sections referenced within this segment.
+    # @return [Array<MachO::Section>] if the Mach-O is 32-bit
+    # @return [Array<MachO::Section64>] if the Mach-O is 64-bit
+    def sections
+      section_klass = case self
+      when MachO::SegmentCommand64
+        MachO::Section64
+      when MachO::SegmentCommand
+        MachO::Section
+      end
+
+      sections = []
+      return sections if nsects.zero?
```

Comment:
You can move this to the very top of the method and make it simply `return [] if nsects.zero?`.